test/e2e-tests.sh

```diff
@@ -108,6 +108,7 @@ restart_pod "${SYSTEM_NAMESPACE}" "app=activator"
 # we need to restart the pod to stop the net-certmanager-controller
 if (( ! HTTPS )); then
   restart_pod "${SYSTEM_NAMESPACE}" "app=controller"
+  kubectl get leases -n "${SYSTEM_NAMESPACE}" -o json | jq -r '.items[] | select(.metadata.name | test("controller.knative.dev.serving.pkg.reconciler.certificate.reconciler")).metadata.name' | xargs kubectl delete lease -n "${SYSTEM_NAMESPACE}"
```

Comment:
There is no controller when we disable internal encryption and when we restart there is no owner.
As I state in the k8s client go bug when the controller shuts down (we have multiple restarts) it does not always clean up stuff. Pls check https://github.com/knative/serving/pull/15321#issuecomment-2176434523 for the details.

Reply:
> The tests in Serving fail because they compare the previous leader set with the new and they find the certificate lease's old owner when they do the intersection of the two sets. As described above this is due to the certificate lease not being updated with the empty holder identity when controller shuts down.


Cool got it - that's what I was missing - can you add that as a comment explaining that

Reply:
> check https://github.com/skonto/test-k8s (README there shows exactly this).

FWIW playing with this it seems like client-side throttling is the culprit - if you disable it then the updates go through.

In our shared main we increase the QPS and Burst based on the number of reconcilers we have

Reply:
In the repo I have you can see:
```
E0610 15:07:15.076863       1 leaderelection.go:308] Failed to release lock: Operation cannot be fulfilled on leases.coordination.k8s.io "lease-70": the object has been modified; please apply your changes to the latest version and try again
```
I am refering to that. That failure is not because of that.

Reply:
@dprotaso added the comment in the description.